felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1031,6 +1040,10 @@ func (m *bpfEndpointManager) updateIfaceStateMap(name string, iface *bpfInterfac
 			iface.dpState.v6.policyIdx[hook.Egress],
 			iface.dpState.filterIdx[hook.Ingress],
 			iface.dpState.filterIdx[hook.Egress],
+			iface.dpState.qosInfo.packetRateTokens[hook.Ingress],
+			iface.dpState.qosInfo.packetRateTokens[hook.Egress],
+			iface.dpState.qosInfo.packetLastTime[hook.Ingress],
+			iface.dpState.qosInfo.packetLastTime[hook.Egress],
```

Comment:
I think we will overwrite the tokens and the timestamp, when felix restarts and programs are attached again. Better to validate this with UT. When we update, its better to read the timestamp and tokens from the map and write the same thing back.

Reply:
Right, I've added reading the QoS state (when on a workload) to the bpfEndpointManager and added coverage for it in the attach_test TestAttachWithMultipleWorkloadUpdate ut. Thanks for catching this!